src/Ctype/Ctype.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Ctype;
+
+/**
+ * Ctype implementation through regex.
+ *
+ * @author Gert de Pagter<BackEndTea@gmail.com
+ */
+final class Ctype
+{
+    /**
+     * Returns TRUE if every character in text is either a letter or a digit, FALSE otherwise.
+     *
+     * @see http://php.net/manual/en/function.ctype-alnum.php
+     *
+     * @param string|int $text
+     *
+     * @return bool
+     */
+    public static function ctype_alnum($text)
+    {
+        return !empty($text) &&
+            !preg_match(
+                '/[^A-Za-z0-9]/',
+                self::convert_int_to_char_for_ctype($text)
+            );
```

Comment:
I know this is minor, but for consistency, could you change all `preg_match()` calls to be one-liners instead?